### PR TITLE
Raise ValidationError if "math" without "math_property" passed to BE from FE

### PR DESCRIPTION
**Pull request by macobo**
BE side fix for https://github.com/PostHog/posthog/issues/5756

I'm not really happy with the experience side of this though, but it's unchanged from current:

https://user-images.githubusercontent.com/148820/132831008-a07234e3-5640-489c-b210-991ed364f83b.mp4

Worth showing a different screen on validation errors (400)?

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

File: ee/clickhouse/queries/trends/util.py [hunk 1]
```diff
@@ -30,9 +30,10 @@ def process_math(entity: Entity) -> Tuple[str, str, Dict[str, Optional[str]]]:
         join_condition = EVENT_JOIN_PERSON_SQL
         aggregate_operation = "count(DISTINCT person_id)"
     elif entity.math in MATH_FUNCTIONS:
-        value, _ = get_property_string_expr(
-            "events", cast(str, entity.math_property), f"%(e_{entity.index}_math)s", "properties"
-        )
+        if entity.math_property is None:
+            raise ValidationError(f"Math without math_property passed")
```
Comment (paolodamico) [inline ee/clickhouse/queries/trends/util.py:34]:
```suggestion
            raise ValidationError({"math_property": "This field is required when `math` is set."}, code="required")
```

File: ee/clickhouse/queries/test/test_trends.py [hunk 2]
```diff
@@ -852,3 +853,9 @@ def test_filtering_with_action_props(self):
         )
 
         self.assertEqual(response[0]["count"], 2)
+
+    def test_trends_math_without_math_property(self):
```
Comment (paolodamico) [inline ee/clickhouse/queries/test/test_trends.py:857]: I would ideally have a test case on the API level as you want to test here that the proper response is sent back.

**Review by paolodamico — APPROVED** [review]
lgtm! but 100% agreed that we need to improve the experience here on the frontend (CC @mariusandra)

File: ee/clickhouse/queries/test/test_trends.py (hunk 2 above)
Comment (macobo) [reply ee/clickhouse/queries/test/test_trends.py:857]: Fair point. Trends don't have any api-level tests though, so let's leave it here for now. :)

Comment (neilkakkar) [reply ee/clickhouse/queries/test/test_trends.py:857]: They do: https://github.com/PostHog/posthog/blob/master/posthog/api/test/test_insight.py#L249

And you can add more CH specific ones here: https://github.com/PostHog/posthog/blob/master/ee/clickhouse/views/test/test_clickhouse_insights.py#L23

**Pull request merged**